src/zarr/abc/store.py

```diff
@@ -320,6 +336,13 @@ async def _get_many(
         for req in requests:
             yield (req[0], await self.get(*req))
 
+    def with_path(self, path: str) -> Self:
```

Comment:
```suggestion
    @abstractmethod
    def with_path(self, path: str) -> Self:
```

Or do you think think we can do this in a generic way that applies to all stores?

Reply:
I don't think we can make it generic unfortunately, because each store class might have its own extra attributes. _maybe_ we could use `__getstate__` to get around this?